core/src/main/java/dk/alexandra/fresco/lib/generic/CommitmentComputation.java

```diff
@@ -33,12 +33,28 @@ public Spdz2kCommitmentComputation(ByteSerializer<HashBasedCommitment> commitmen
   public DRes<List<byte[]>> buildComputation(ProtocolBuilderNumeric builder) {
     HashBasedCommitment ownCommitment = new HashBasedCommitment();
     byte[] ownOpening = ownCommitment.commit(localDrbg, value);
-    return builder.seq(new BroadcastComputation<>(commitmentSerializer.serialize(ownCommitment)))
+    return builder.seq(
+        seq -> {
+          if (noOfParties > 2) {
```

Comment:
would be nice to comment here that for <= 2 `InsecureBroadcastProtocol` can be used because there's no other parties to verify against anyway. (I know it "makes sense", but it doesn't hurt to explain)